dataland-external-storage/src/main/kotlin/org/dataland/datalandexternalstorage/services/EurodatSafeDepositBoxInitialization.kt

```diff
@@ -0,0 +1,59 @@
+package org.dataland.datalandexternalstorage.services
+
+import jakarta.annotation.PostConstruct
+import org.dataland.datalandeurodatclient.openApiClient.api.SafeDepositDatabaseResourceApi
+import org.dataland.datalandeurodatclient.openApiClient.model.SafeDepositDatabaseRequest
+import org.dataland.datalandeurodatclient.openApiClient.model.SafeDepositDatabaseResponse
+import org.dataland.datalandexternalstorage.utils.EurodatDataStoreUtils.retryWrapperMethod
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+
+/**
+ * Simple implementation of the initialization of the eurodat minabo safedepositbox
+ * @param safeDepositDatabaseResourceClient the service to create the safe deposit box used to store private data
+ * on eurodat
+ */
+@Component
+class EurodatSafeDepositBoxInitialization(
+    @Autowired var safeDepositDatabaseResourceClient: SafeDepositDatabaseResourceApi,
+    @Value("\${dataland.eurodatclient.app-name}")
+    private val eurodatAppName: String,
+    @Value("\${dataland.eurodatclient.initialize-safe-deposit-box}")
+    private val initializeSafeDepositBox: Boolean,
+
+) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Tries to create a safe deposit box in EuroDaT for storage of Dataland data a pre-defined number of times and
+     * then throws a final exception after the retries are used up.
+     */
+    @PostConstruct
+    fun createSafeDepositBox() {
+        if (initializeSafeDepositBox) {
```

Comment:
Statt hier zu prüfen, ob die Box initialized werden soll, kann man die komplette Komponente hier mit einer Annotation versehen wie hier:
`@ConditionalOnProperty(name = ["dataland.eurodatclient.initialize-safe-deposit-box"], havingValue = "true", matchIfMissing = false)`

Dann erstellt SPring diese Komponente nur dann, wenn `initiliaze-safe-deposit-box` in den app.props auf "true" gesetzt ist.

Reply:
Ok, wenn das funktioniert ändere ich das und pushe das. Ich habe das selbst noch nicht ausprobiert.

Reply:
(Weil im dev-Profil ist das glaube ich auf "false" gesetzt)

Reply:
Hatte getestet, wenn man ihm die property nicht gleich "true" gibt, dann startet er die Box nicht. Scheint also zu funktionieren.